### PR TITLE
fix: unhandled exception or failing error body

### PR DESCRIPTION
**Pull request by ronag**
Refs: https://github.com/nodejs/undici/issues/3136

### CODE REVIEW

**Comment by ronag** [comment]
@sandratatarevicova  Do you think you can help create a test that does not depend on external url?

File: test/issue-3136.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const { request } = require('..')
+
+request('http://www.rolbox.it/rk-kastelruth/webcam/web03.jpg', { throwOnError: true })
```
Comment (mcollina) [inline test/issue-3136.js:3]: This needs a server, we can't use an external url

**Review by mcollina — CHANGES_REQUESTED** [review]

**Review by metcoder95 — COMMENTED** [review]
LGTM, just waiting for tests around it

**Comment by sandratatarevicova** [comment]
> @sandratatarevicova Do you think you can help create a test that does not depend on external url?

Thank you! I have modified the test in https://github.com/sandratatarevicova/undici/commit/d9183af7357517d91325315165c49a9330823f27. Should I create a PR to your branch?

**Comment by ronag** [comment]
@sandratatarevicova if you don't min I'll just copy it into this PR

**Review by mcollina — APPROVED** [review]
LGTM

**Review by metcoder95 — APPROVED** [review]

**Pull request merged**